File: Code/Chapter 8/C#/CreateDataSet/DataSetFiller.cs
#region Using directives

using System;
using System.Collections.Generic;
using System.Text;
using System.Data;

#endregion

namespace CreateDataSet
{
   public static class DataSetFiller
   {
      public static DataSet FillDataset(string xmlFile)
      {
         DataSet ds = new DataSet();
         ds.ReadXml(xmlFile);
         return ds;
      }

      public static CustProd FillStrongDataSet(string xmlFile)
      {
         CustProd ds = new CustProd();
         ds.ReadXml(xmlFile);
         return ds;
      }
   }
}

File: Code/Chapter 8/VB.NET/CreateDataSet/DataSetFiller.vb
Imports System
Imports System.Data

Public Class DataSetFiller
   Public Shared Function FillDataset(ByVal xmlFile As String) As DataSet
      Dim ds As New DataSet()
      ds.ReadXml(xmlFile)
      Return ds
   End Function

   Public Shared Function FillStrongDataSet(ByVal xmlFile As String) As CustProd
      Dim ds As New CustProd()
      ds.ReadXml(xmlFile)
      Return ds
   End Function
End Class
